include/mxnet/c_api.h

```diff
@@ -321,10 +321,31 @@ MXNET_DLL int MXNDArraySlice(NDArrayHandle handle,
                              mx_uint slice_end,
                              NDArrayHandle *out);
 /*!
+ * \brief Slice the NDArray along any axis.
+ * \param handle the handle to the ndarray
+ * \param axis the axis along which to do slicing
+ * \param slice_begin The beginning index of slice
+ * \param slice_end The ending index of slice
+ * \param out The NDArrayHandle of the output NDArray
+ * \return 0 when success, -1 when failure happens
+ *
+ * Unlike the MXNDArraySlice method in which the sliced
+ * array shares memory with the original array. This
+ * method always copies the data.
+ *
+ * Also this method expects that out is pre-allocated with
+ * appropriate shape.
+ */
+MXNET_DLL int MXNDArrayCopySliceTo(NDArrayHandle handle,
```

Comment:
see how copyto is redirected into an ndarray operator https://github.com/dmlc/mxnet/blob/master/python/mxnet/ndarray.py#L472